streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroStreamsSnapshotDeserializer.java

```diff
@@ -65,7 +65,10 @@ private Object convertValue(Object deserializedObj) {
             List<Schema.Field> fields = indexedRecord.getSchema().getFields();
             ImmutableMap.Builder<String, Object> keyValues = ImmutableMap.builder();
             for (Schema.Field field : fields) {
-                keyValues.put(field.name(), convertValue(indexedRecord.get(field.pos())));
+                Object currentValue = convertValue(indexedRecord.get(field.pos()));
+                if (currentValue != null) {
```

Comment:
@priyank5485 @satishd @arunmahadevan changing this for minor release seems to be risky, especially changing the immutable map.  Lets file another issue for next major release to keep track of it.
Overall I think @arunmahadevan suggestion looks good to use `Optional.empty()`. I don't think we've clear answer how better to deal with it. On one hand removing the key when value is null breaks the schema contract but having them in there means Users in their code needs to handle this. 
Especially incase of generic UDFs/CPs they have to write the code defensively as they don't know which schema's field might be null.